crates/common/validator/src/beacon_api_client/mod.rs

```diff
@@ -73,10 +75,34 @@ impl BeaconApiClient {
             .boxed())
     }
 
+    pub async fn get_node_syncing_status(
+        &self,
+    ) -> anyhow::Result<DataResponse<SyncStatus>, ValidatorError> {
+        let response = self
+            .http_client
+            .execute(
+                self.http_client
+                    .get("/eth/v1/node/syncing".to_string())?
+                    .build()?,
+            )
+            .await?;
+
+        if !response.status().is_success() {
+            return Err(ValidatorError::RequestFailed {
+                status_code: response.status(),
+            });
+        }
+
+        response
+            .json()
+            .await
+            .map_err(|err| ValidatorError::JsonDecodeError(err.to_string()))
```

Comment:
```suggestion
        response
            .json()
            .await
```

we should just be able to do this no? as from should be implemented right?

Reply:
fix this in other places as well